src/extensions/default/CodeFolding/foldhelpers/handlebarsFold.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * Fold range finder for handlebars/mustache template type files.
+ * @author Patrick Oladimeji
+ * @license MIT
+ * @date 14/08/2016 22:04:21
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+    var CodeMirror = brackets.getModule("thirdparty/CodeMirror/lib/codemirror"),
+        space = /\s/;
+    /**
+     * Utility function for scanning the text in a document until a certain condition is met
+     * @param {object}   cm        The code mirror object representing the document
+     * @param {string}   startCh   The start character position for the scan operation
+     * @param {number}   startLine The start line position for the scan operation
+     * @param {function} condition A predicate function that takes in the text seen so far
+     */
+    function scanTextUntil(cm, startCh, startLine, condition) {
+        var line = cm.getLine(startLine);
+        var seen = '',
+            characterIndex = startCh,
+            currentLine = startLine;
+        while (currentLine <= cm.lastLine()) {
+            if (line.length === 0) {
+                characterIndex = 0;
+                line = cm.getLine(++currentLine);
+            } else {
+                seen = seen.concat(line[characterIndex] || '');
+                if (condition(seen)) {
+                    return {from: {ch: startCh, line: startLine}, to: {ch: characterIndex, line: currentLine}, string: seen};
+                } else if (characterIndex >= line.length) {
+                    seen = seen.concat(cm.lineSeparator());
+                    if (condition(seen)) {
+                        return {from: {ch: startCh, line: startLine}, to: {ch: characterIndex, line: currentLine}, string: seen};
+                    }
+                    characterIndex = 0;
+                    line = cm.getLine(++currentLine);
+                } else {
+                    ++characterIndex;
+                }
+            }
+        }
+    }
+    
+    function endTag(seen) {
+        return space.test(seen[seen.length - 1]) || seen[seen.length - 1] === '}';
```

Comment:
Sorry, I think it easy to forget the -1.
So having something:
```js
function lastchar(string) {
  return string[string.length - 1]
}
function endTag(seen) {
  return space.test(lastchar(seen)) || lastchar(seen) === '}';
}
```
or maybe leave the first condition as is and use
```js
StringUtils.endsWith(seen, "}");
```
on the second condition and elsewhere.
```js
function endTag(seen) {
  return space.test(seen[seen.length - 1]) || StringUtils.endsWith(seen, "}");
}
```